book/src/fullnode-with-ledger-notifications.md

```diff
@@ -0,0 +1,35 @@
+# Fullnode
+
+This document proposes the following architecture changes to fullnode pipeline. 
+
+## Current Architecture
+<img alt="Fullnode block diagrams" src="img/fullnode.svg" class="center"/>
+
+## Proposal
+<img alt="Fullnode block diagrams" src="img/fullnode-proposal.svg" class="center"/>
+
+## Background
+
+In the old architecture, the ReplayStage receives entries from the RetransmitStage, 
+processes them and writes to the ledger.
+
+In the current code, the ledger write operation has been moved out of the ReplayStage.
+If the fullnode is running with a TPU, the Broadcast Service writes to the ledger.
+The Retransmit Stage receives entries from peer fullnodes and also writes to the ledger.
+
+The ledger notifies the ReplayStage whenever some entity writes to it. The 
+ReplayStage reads from the ledger, processes the entries in the bank and submits
+the votes.
+
+In old architecture, the RetransmitStage and ReplayStage are contained in the TVU.
+
+## Changes
+
+This document proposes the following change to the architecture.
+
+1. Split TVU into Peer Transaction Receiver (contains BlobFetchStage, and RetransmitStage)
+2. Add Blocktree (ledger) to the diagram
+3. The Blocktree is updated by Broadcast Stage and Peer Transaction Receiver
```

Comment:
I really regret changing the name Broadcast Stage to Broadcast Service, but until we change it back, let's use the term from the codebase, Broadcast Service.

Reply:
👍